### PR TITLE
Add Excel (.xlsx) import strategy support

### PR DESCRIPTION
**Pull request by NikolayMetchev**
## Summary

Crypto.com's older "Card Transaction History" export is an Excel workbook in a different layout than their usual CSV files, so a dedicated strategy type is needed. Rather than build a parallel pipeline, XLSX rides the existing CSV strategy engine end to end (the same way QIF does): an `.xlsx` worksheet is parsed into the same headers/rows shape a CSV produces, then flows through the unchanged `StrategySelector`, `CsvTransferMapper`, `CsvImportApplier`, and re-import machinery.

- New `utils/parsers/xlsx` module: Apache POI on the JVM behind an expect/actual seam (Android reports the platform unsupported — POI is JVM-only, not Android-friendly)
- Binary file picker (`utils/compose/filePicker`) alongside the existing text-only one, since decoding `.xlsx` bytes as UTF-8 would corrupt the file
- `xlsx_import_blob` table stores the raw workbook bytes next to the existing staged-row table, so a strategy naming a different worksheet can re-extract it later
- `CsvImportStrategy.worksheetName` + `StrategyKind.XLSX` thread an Excel strategy through the catalog/sync codecs the same way QIF does
- `ImportDirectoryScanner` stages `.xlsx` files (JVM-only; Android records a per-file scan failure rather than crashing)
- Built-in "Crypto.com Card (Excel)" strategy, tuned against the real workbook: signed `Amount Processed` for direction, `Card Acceptor Name` trimmed of its fixed-width city/country padding for both the account name and Curve pass-through detection, and a Cash-account fallback for rows with no card acceptor name (e.g. "Batch Credit Funds Transfer")

## Test plan
- [x] `./gradlew build` passes locally (all modules, all tests)
- [x] `./gradlew buildHealth` passes
- [x] New unit tests: `XlsxParserTest` (parsing, multi-sheet selection, edge cases) and `CryptoComCardXlsxMapperTest` (field mapping, Curve pass-through routing, blank-acceptor-name handling) against real row shapes
- [x] Verified against the real workbook via the default local database: import succeeds, Curve pass-through correctly collapses `CRV*` merchant rows instead of creating junk accounts, and the two previously-failing "Batch Credit Funds Transfer" rows now resolve to the Cash account

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Import and stage `.xlsx` files from the Imports screen, selecting a worksheet and converting it into importable rows.
  * Save and manage worksheet-specific Excel import strategies, including Excel labeling and filtering in the strategy catalog.
  * Added a binary file picker to read workbook bytes safely.
* **Bug Fixes**
  * Improved duplicate detection by hashing raw workbook bytes for Excel imports; preserved original workbook data for later worksheet re-extraction.
* **Tests**
  * Added XLSX parser and Crypto.com Card Excel strategy test coverage.
* **Platform Support**
  * Excel parsing is available on desktop; Android reports XLSX imports as unsupported.

### CODE REVIEW

File: app/db/schema/src/commonMain/sqldelight/com/moneymanager/database/sql/csvImportStrategy/CsvImportStrategy.sq [hunk 1]
```diff
@@ -13,6 +13,7 @@ CREATE TABLE csv_import_strategy (
     cross_source_reconcile_window_seconds INTEGER,
     conversion_config_json TEXT,
     funding_attribute_match_json TEXT,
+    worksheet_name TEXT,
```
Comment (NikolayMetchev) [inline app/db/schema/src/commonMain/sqldelight/com/moneymanager/database/sql/csvImportStrategy/CsvImportStrategy.sq]: do we need this column if it is already in xlsx_import_blob

Comment (NikolayMetchev) [reply app/db/schema/src/commonMain/sqldelight/com/moneymanager/database/sql/csvImportStrategy/CsvImportStrategy.sq]: Good point — this wasn't just an optional feature like the other nullable columns on this table (file_name_pattern, conversion_config, funding_attribute_match, which any CSV strategy might opt into); worksheet_name is the XLSX-vs-CSV/QIF kind discriminator, so it deserved the same satellite-table treatment as xlsx_import_blob. Split it into a new `xlsx_import_strategy` table (1:1 via csv_import_strategy_id, cascade-deleted with the parent) in 0ca18822. csv_import_strategy no longer has a worksheet_name column.
